### ISSUE TITLE
Found array with dim 3. check_pairwise_arrays expected <= 2

### ISSUE THREAD

**Issue by Krukiou** [MaartenGr/BERTopic#1569]
Hi,

Bertopic is generating the error in the title of this issue.
Here an extract of the code I'm using : 
```
from nltk.stem import WordNetLemmatizer
import nltk
import re
import logging
import pickle
from sklearn.pipeline import make_pipeline
from sklearn.decomposition import TruncatedSVD
from sklearn.feature_extraction.text import TfidfVectorizer

from nltk.corpus import stopwords
from nltk import word_tokenize

from scipy.cluster import hierarchy as sch
from hdbscan import HDBSCAN
from umap import UMAP
from bertopic import BERTopic
#from sentence_transformers import SentenceTransformer

logging.basicConfig(level=logging.INFO)

# nltk.download('punkt')
# nltk.download('wordnet')
# nltk.download('stopwords')

wn = WordNetLemmatizer()

stopwords_list = stopwords.words('french')
stopwords_list.extend(['être', 'avoir', 'celui', 'celle', 'ceux', 'celles', 'donc', 'puisque', 'ainsi', 'alors'])
final_stopwords_list = stopwords_list

def reclean(d):
    newd = ''
    for dd in d.split(' '):
        if dd.isalpha():
            newd = newd + ' ' + dd
    return newd  

# def reclean(d):
#     cleaned_sentences = []
#     for sentence in d:  # For each sublist
#         cleaned_sentence = ' '.join([word.lower() for word in sentence if word.isalpha()])
#         cleaned_sentences.append(cleaned_sentence)
#     return cleaned_sentences

def preprocess_data(data):
    print(type(data['text'][0]))
    print(data['text'][0])
    print(len(data['text'][0]))

    #Apply reclean to data['text']
    data['text'] = [reclean(text) for text in data['text']]

    print(type(data['text'][0]))
    
    # Flatten the list of lists into a single list of cleaned sentences
    #flattened_texts = [' '.join(text) for text in data['text']]

    data['text'] = [' '.join([word.lower() for word in word_tokenize(text) if word.lower() not in final_stopwords_list]) for text in data['text']]
    data['text'] = [' '.join([wn.lemmatize(w) for w in word_tokenize(text) if w not in final_stopwords_list]) for text in data['text']]
    #drop na in data['text']
    data = data.dropna(subset=['text'])

    return data

def create_vectorizer():
    vectorizer = TfidfVectorizer(ngram_range=(1,2),
                            tokenizer=nltk.word_tokenize,
                            lowercase =True, analyzer='word',
                            stop_words = final_stopwords_list,
                            max_features = 20000)
    return vectorizer

def create_topic_model():
    umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=100)
    hdbscan_model = HDBSCAN(min_cluster_size=30, metric='euclidean', cluster_selection_method='eom',min_samples=30, prediction_data=True)
    topic_model = BERTopic(language="french", calculate_probabilities=True, nr_topics=50,
                            umap_model=umap_model,
                            hdbscan_model=hdbscan_model)
    return topic_model

def bert_clustering_topics_modelling(data, model_directory, version):

    logging.info('starting preprocessing data')
    try:
        data = preprocess_data(data)
    except Exception as e:
        logging.error('error during preprocess data - %s'%(e))
    
    logging.info('starting create vectorizer')
    try:    
        vectorizer = create_vectorizer()
    except Exception as e:
        logging.error('error during create vectorizer - %s'%(e))
    
    logging.info('starting create topic modelling')
    try:
        topic_model = create_topic_model()
    except Exception as e:
        logging.error('error during create topic model- %s'%(e))
    
    logging.info('starting vectorizer fit transform')
    try:
        #sample_data = data.head(10)
        #sample_data['text'] = sample_data['text'].astype(str)
        #print(type(sample_data['text']))
        embeddings = vectorizer.fit_transform(data['text'])
        # with open(model_directory+'/vectorizer_'+str(version)+'.pickle', 'wb') as handle:
        #     pickle.dump(vectorizer, handle, protocol=pickle.HIGHEST_PROTOCOL)
    except Exception as e:
        logging.error('error during vectorizer fit transform - %s'%(e))
    
    logging.info('starting topic model fit transform')
    try:

        topics, probs = topic_model.fit_transform(data['text'],embeddings=embeddings)
        data['topic'] = topics
        #hierarchical_topics = topic_model.hierarchical_topics(data['text'])
        #tree = topic_model.get_topic_tree(hierarchical_topics)
        #fig = topic_model.visualize_hierarchy(hierarchical_topics=hierarchical_topics, width=2000)
        # with open(model_directory+'/topic_model_'+str(version)+'.pickle', 'wb') as handle:
        #     pickle.dump(topic_model, handle, protocol=pickle.HIGHEST_PROTOCOL)
    except Exception as e:
        logging.error('error during topic model fit transform - %s'%(e))

    # Prepare topic_dict
    vectorizer = vectorizer
    topic_model = topic_model
    topic_dict = {}
    topics_list = topic_model.topics_
    for i in topics_list : 
        most_representatif_word = topic_model.get_topic(i)[0][0]
        topic_dict["cluster_" + str(i)] = [x[0] for x in topic_model.get_topic(i)[:5]]



    return data, vectorizer, topic_model, topic_dict


```
I checked the shape of the variables : 
data['text'] : shape = (6000,)
embeddings : shape = (6000,18751)

When all these data are submitted to Bertopic, Bertopic is generating this error

When I'm considering only the first 100 rows of data['text], the code is working fine.
In this last case we have : 
data['text'] : shape = (100,)
embeddings : shape = (100,1650)

thus, from my point of view, the error comes from the data.
What can be the root cause of this error ?
How can I debug this code - find the root cause and correct my code ?

Thanks you, @MaartenGr

**Comment by MaartenGr** [comment]
> How can I debug this code - find the root cause and correct my code ?

You can start by doing the following. First, check the type hints of your code. BERTopic expects a list of strings and not a pandas series, so changing that might help. Second, remove parameters one at a time to see whether one of these had some issues. For example, I see that you are using a custom tokenizer which might not be initialized correctly. 

Lastly, make sure to also show the complete error. Currently, I cannot see what the source of the issue is without the full log.

**Comment by Krukiou** [comment]
Hi, 
The problem came from my embedding indeed : replacing my current vectorizer by the the default vectorizer of Bertopic has solved this problem.
Thank you @MaartenGr
